python/pyspark/sql/connect/session.py

```diff
@@ -326,6 +327,28 @@ def readStream(self) -> "DataStreamReader":
 
     readStream.__doc__ = PySparkSession.readStream.__doc__
 
+    def registerProgressHandler(self, handler: "ProgressHandler") -> None:
+        """
+        Register a progress handler to be called when a progress update is received from the server.
+        .. versionadded:: 4.0
```

Comment:
API should follow https://numpydoc.readthedocs.io/en/latest/format.html with describing return type, and example if possible.